File: C/Code/4600/4673 - 셀프 넘버.c
#include <stdio.h>

int arr[10100];

void d(int n) {
	int temp = n;
	while (n)
		temp += n % 10, n /= 10;
	arr[temp] = 1;
	return;
}

int main() {
	for (int i = 1; i < 10000; i++) {
		// find i's constructor
		d(i);

		// if self number
		if (!arr[i])
			printf("%d\n", i);
	}
	return 0;
}

File: Python/Code/4600/4673 - 셀프 넘버.py
notSelf = []
for i in range(1, 10001):
	a = i
	temp = i
	while a > 0:
		temp += a % 10
		a //= 10
	notSelf.append(temp)
for i in range(1, 10001):
	if i not in notSelf:
		print(i)